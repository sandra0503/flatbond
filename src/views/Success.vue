<template>
  <div class="success">
    <router-link to="/" class="back-link"
      ><i class="arrow-left" />Back to previous page</router-link
    >
    <div class="success__content">
      <h1>Thank you!</h1>
      <div>Your flatbond was created with the following details:</div>
      <table class="details-table">
        <tr>
          <th class="label">Monthly rent:</th>
          <td class="value rentValue">{{ formatMonthlyRent }} £</td>
        </tr>
        <tr>
          <th class="label">Postcode:</th>
          <td class="value postcodeValue">{{ postcode }}</td>
        </tr>
        <tr>
          <th class="label">Membership fee:</th>
          <td class="value feeValue">{{ fee }} £</td>
        </tr>
      </table>
    </div>
    <Illustration :imageSrc="require('@/assets/home.png')" position="right" />
  </div>
</template>

<script>
import { mapState } from "vuex";
import Vue from "vue";
import Router from "vue-router";
import Illustration from "@/components/Illustration.vue";

Vue.use(Router);

export default {
  name: "success",
  components: {
    Illustration
  },
  computed: {
    ...mapState(["rent", "postcode", "fee"]),
    formatMonthlyRent() {
      return (this.rent / 100).toFixed(2);
    }
  }
};
</script>

<style lang="scss" scoped>
.success {
  text-align: center;
  max-width: 1024px;
  margin: 0 auto;
}

.success__content {
  padding: 30px 0 250px;
}

.back-link {
  width: 100%;
  display: inline-block;
  text-align: left;
  color: var(--blue);
  text-decoration: none;
  font-weight: 500;

  &:hover {
    color: #4166ff;
  }
}

.arrow-left {
  border: solid var(--blue);
  border-width: 0 2px 2px 0;
  display: inline-block;
  padding: 3px;
  margin-right: 4px;
  transform: rotate(135deg);
  -webkit-transform: rotate(135deg);
}

.details-table {
  margin: 25px auto;

  .label,
  .value {
    width: 50%;
    text-align: right;
    padding: 5px;
  }

  .value {
    text-align: left;
  }
}
</style>
