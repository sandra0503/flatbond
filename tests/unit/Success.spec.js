import { shallowMount, createLocalVue } from "@vue/test-utils";
import Success from "@/views/Success";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Success view", () => {
  let state;

  const build = () => {
    const wrapper = shallowMount(Success, {
      localVue,
      store: new Vuex.Store({
        state
      })
    });

    return {
      wrapper,
      rentValue: () => wrapper.find(".rentValue"),
      postcodeValue: () => wrapper.find(".postcodeValue"),
      feeValue: () => wrapper.find(".feeValue")
    };
  };

  beforeEach(() => {
    state = { rent: 20000, postcode: "WC2N 5DU", fee: 144 };
  });

  it("renders the component and matches snapshot", () => {
    const { wrapper } = build();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main content elements", () => {
    const { rentValue, postcodeValue, feeValue } = build();
    expect(rentValue().exists()).toBe(true);
    expect(postcodeValue().exists()).toBe(true);
    expect(feeValue().exists()).toBe(true);
  });

  it("formats the rent value correctly", () => {
    const { rentValue } = build();
    expect(rentValue().text()).toBe("200.00 Â£");
  });
});
