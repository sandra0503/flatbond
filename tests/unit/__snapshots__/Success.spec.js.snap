// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success view renders the component and matches snapshot 1`] = `
<div class="success">
  <router-link-stub to="/" tag="a" event="click" class="back-link"><i class="arrow-left"></i>Back to previous page</router-link-stub>
  <div class="success__content">
    <h1>Thank you!</h1>
    <div>Your flatbond was created with the following details:</div>
    <table class="details-table">
      <tr>
        <th class="label">Monthly rent:</th>
        <td class="value rentValue">200.00 £</td>
      </tr>
      <tr>
        <th class="label">Postcode:</th>
        <td class="value postcodeValue">WC2N 5DU</td>
      </tr>
      <tr>
        <th class="label">Membership fee:</th>
        <td class="value feeValue">144 £</td>
      </tr>
    </table>
  </div>
  <illustration-stub imagesrc="" position="right"></illustration-stub>
</div>
`;
